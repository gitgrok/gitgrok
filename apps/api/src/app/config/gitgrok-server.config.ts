import { Injectable } from '@nestjs/common';
import { Env, Parameter } from '@onivoro/server-parameterization';
import { resolve } from 'path';

@Env(String, 'GITGROK_MANIFEST')
export class ManifestPath extends Parameter<string> {
  value = resolve('~/.gitgrok/.manifest');
}
@Env(String, 'GITGROK_HOME')
export class HomePath extends Parameter<string> {
  value = resolve('~/.gitgrok');
}
@Env(String, 'GITGROK_PORT')
export class GitGrokPort extends Parameter<string> {
  value = 7777;
}

@Injectable()
export class GitgrokServerConfig {
  constructor(
    public readonly manifestPath: ManifestPath,
    public readonly homePath: HomePath,
    public readonly port: GitGrokPort
  ) {}
}
