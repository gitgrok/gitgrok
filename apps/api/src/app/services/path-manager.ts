import { Injectable } from '@nestjs/common';
import { parse } from 'path';
import { HomePath, ManifestPath } from '../config/gitgrok-server.config';

@Injectable()
export class PathManager {
  constructor(
    private readonly homePath: HomePath,
    private readonly manifestPath: ManifestPath
  ) {}

  getManifestPath() {
    return this.manifestPath;
  }

  getHomeDirectory(scmProviderPath?: string): string {
    return `${this.homePath.value}/${scmProviderPath || ''}/`.replace(
      '//',
      '/'
    );
  }

  extractDirToConeInFromUrl(url: string): string {
    return parse(`${this.homePath.value}/${url.split('https://')[1]}`).dir;
  }

  extractProjectDirFromUrl(url: string): string {
    const dir = this.extractDirToConeInFromUrl(url);
    const segments = url.split('/');
    return `${dir}/${segments[segments.length - 1].replace('.git', '')}`;
  }
}
