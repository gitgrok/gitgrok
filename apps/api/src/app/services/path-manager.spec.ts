import { Injectable } from '@nestjs/common';
import { parse } from 'path';
import { HomePath, ManifestPath } from '../config/gitgrok-server.config';
import { PathManager } from './path-manager';
const ng = 'https://github.com/angular/angular-cli.git';
const homePath: HomePath = { value: 'HOMEPATH' } as any;
const manifestPath: ManifestPath = { value: 'MANIFESTPATH' } as any;

describe(PathManager.name, () => {
  describe(PathManager.prototype.extractProjectDirFromUrl.name, () => {
    it.each([[ng]])(
      PathManager.prototype.extractProjectDirFromUrl.name,
      (url: string) => {
        expect(
          new PathManager(homePath, manifestPath).extractProjectDirFromUrl(url)
        ).toMatchSnapshot();
      }
    );
  });

  describe(PathManager.prototype.extractDirToConeInFromUrl.name, () => {
    it.each([[ng]])(
      PathManager.prototype.extractProjectDirFromUrl.name,
      (url: string) => {
        expect(
          new PathManager(homePath, manifestPath).extractDirToConeInFromUrl(url)
        ).toMatchSnapshot();
      }
    );
  });
});
