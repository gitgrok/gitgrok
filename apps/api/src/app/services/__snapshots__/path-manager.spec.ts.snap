// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PathManager extractDirToConeInFromUrl extractProjectDirFromUrl 1`] = `"HOMEPATH/github.com/angular"`;

exports[`PathManager extractProjectDirFromUrl extractProjectDirFromUrl 1`] = `"HOMEPATH/github.com/angular/angular-cli"`;
